packages/contracts-sdk/src/lib/contracts-sdk.ts

```diff
@@ -2394,6 +2404,23 @@ https://developer.litprotocol.com/v3/sdk/wallets/auth-methods/#auth-method-scope
       }: MintNextAndAddAuthMethods): Promise<any> => {
         // first get mint cost
         const mintCost = await this.pkpNftContract.read.mintCost();
+
+        let _gasLimit: BigNumberish | undefined = gasLimit;
+        if (!_gasLimit) {
+          const txData =
+            await this.pkpHelperContract.write.populateTransaction.mintNextAndAddAuthMethods(
```

Comment:
As above, it would be good to get this defined as its own function rather than having to define all of the transaction arguments twice and use a local variable to store gasLImit

Reply:
Responded above